sympy/tensor/array/dense_ndim_array.py

```diff
@@ -164,7 +179,7 @@ def _new(cls, iterable, shape, **kwargs):
         self._shape = shape
         self._array = list(flat_list)
         self._rank = len(shape)
-        self._loop_size = functools.reduce(lambda x,y: x*y, shape) if shape else 0
+        self._loop_size = functools.reduce(lambda x,y: x*y, shape) if shape else len(flat_list)
```

Comment:
why this?

Reply:
This is because in the old way, an Array with one item will have a `_loop_size` equal to 0. I have opened an issue about it: #17230. Maybe there is a reason for keeping the `_loop_size` as 0? If so, I would be happy to know it. :-)

Reply:
If `shape == ()`, maybe loop size should be 1? (i.e. it's a scalar)